sky/execution.py

```diff
@@ -203,18 +206,31 @@ def _execute(
                 f'Backend {backend.NAME} does not support autostop, please try '
                 f'{backends.CloudVmRayBackend.NAME}')
 
-    if not cluster_exists and Stage.OPTIMIZE in stages:
-        if task.best_resources is None:
-            # TODO: fix this for the situation where number of requested
-            # accelerators is not an integer.
-            if isinstance(backend, backends.CloudVmRayBackend):
-                # TODO: adding this check because docker backend on a
-                # no-credential machine should not enter optimize(), which
-                # would directly error out ('No cloud is enabled...').  Fix by
-                # moving `sky check` checks out of optimize()?
-                dag = sky.optimize(dag, minimize=optimize_target)
-                task = dag.tasks[0]  # Keep: dag may have been deep-copied.
-                assert task.best_resources is not None, task
+    if not cluster_exists:
+        if (Stage.PROVISION in stages and task.use_spot and
+                not _is_launched_by_spot_controller):
+            yellow = colorama.Fore.YELLOW
```

Comment:
random: Would it be a good idea to have a `sky.colors` and `from sky.colors import *` in our code since these colors are used quite a lot (and then maybe instead of specifying yellow it could be `sky.colors.warning_color`. Obviously not very relevant to this PR, but just a thought.

Reply:
+1, with the suggestion that we should avoid adding new modules if possible - perhaps add to existing `log_utils` or `sky_logging`.

Example interface: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/cli_logger.py#L552